### PR TITLE
Single regressor

### PR DESCRIPTION
**Pull request by arokem**
This PR is intended to address the issues raised in #24. 

@oliver-contier : does this test replicate what you are trying to do? 

Locally, I don't see the error that you mentioned with the first of the two commits (https://github.com/nrdg/fracridge/pull/25/commits/6a053a3c3d3d05497e56d5972f6db7de2b853487), but I do get a similar error on the second commit (https://github.com/nrdg/fracridge/pull/25/commits/bc88548b4c254c155e204db9dbc19a97aa066991). Do you get that error on calls to `fit`? Or only on a subsequent call to `predict`?

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -1,3 +1,13 @@
+1.3.2 (February 3rd, 2021)
+==========================
+Fixes a bug in cases where the number of targets is larger than
+the number of samples in each target.
+
+1.3 (December 2, 2020)
+=========================
+This version includes changes to the documentation, to refer to the
+published paper.
+
```
Comment (arokem) [inline CHANGES.rst:10]: These changes are unrelated to the main purpose of this PR, but may as well merge them here as well.

**Comment by arokem** [comment]
OK - I think that the last commit addresses this special case for the `predict` method as well. 

@oliver-contier : if you could give this a try with your use-case, that would be helpful!

**Comment by oliver-contier** [comment]
Strange, I still get the same error when calling `fit`:

```
fracridge_error ❯ ipython
Python 3.8.5 | packaged by conda-forge | (default, Sep 24 2020, 16:37:41)
Type 'copyright', 'credits' or 'license' for more information
IPython 7.23.1 -- An enhanced Interactive Python. Type '?' for help.

In [1]: from fracridge import FracRidgeRegressor
   ...: import numpy as np
   ...:
   ...: x_train = np.random.rand(312, 1)   # (312 samples, 1 regressor) 
   ...: y_train = np.random.rand(312, 211)  # (312 samples, 211 targets)
   ...:
   ...: fracs_ = np.arange(0., 1., .05)
   ...:
   ...: fr = FracRidgeRegressor(fracs=fracs_, fit_intercept=True, normalize=True)
   ...: fr.fit(x_train, y_train)
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-1-37edc04e7a92> in <module>
      8
      9 fr = FracRidgeRegressor(fracs=fracs_, fit_intercept=True, normalize=True)
---> 10 fr.fit(x_train, y_train)

~/Desktop/try_fracridge/fracridge/fracridge/fracridge.py in fit(self, X, y, sample_weight)
    310         self.alpha_ = alpha
    311         self.coef_ = coef
--> 312         self._set_intercept(X_offset, y_offset, X_scale)
    313         self.is_fitted_ = True
    314         return self

~/Desktop/try_fracridge/fracridge/fracridge/fracridge.py in _set_intercept(self, X_offset, y_offset, X_scale)
    334             else:
    335                 self.coef_ = self.coef_ / X_scale[:, np.newaxis, np.newaxis]
--> 336             self.intercept_ = y_offset - np.tensordot(X_offset,
    337                                                       self.coef_, axes=(1))
    338         else:

<__array_function__ internals> in tensordot(*args, **kwargs)

/opt/anaconda3/envs/fracridge_error/lib/python3.8/site-packages/numpy/core/numeric.py in tensordot(a, b, axes)
   1114                 axes_b[k] += ndb
   1115     if not equal:
-> 1116         raise ValueError("shape-mismatch for sum")
   1117
   1118     # Move the axes to sum over to the end of "a"

ValueError: shape-mismatch for sum
```

Having a look at the traceback, it seems to have something to do with the `np.tensordot` call in `_set_intercept`? I'm sorry I didn't notice this sooner! Indeed, when passing `fit_intercept=False` in creating the `FracRidgeRegressor` object leads to no error. Might it have something to do with the `axes=(1)` argument in the call to `np.tensordot`?

```
fr = FracRidgeRegressor(fracs=fracs_, fit_intercept=False, normalize=True)
fr.fit(x_train, y_train)
# works fine
```

**Comment by kendrickkay** [comment]
Sorry I don't have insight into the Python implementation.  Perhaps a little hack like including a column of zeros in design matrix might do the trick for now until this can be sorted out?  (Assuming the code handles that corner case well.)  I can vouch for the matlab version of fracridge :)

**Comment by arokem** [comment]
@oliver-contier : I think that you were spot on -- the `axes` argument seems to have done the trick. Does this work for you?

**Pull request merged**